telegram中的MTProto代理不支持fake tls

### 路由地址

```routes
/telegram/channel/:username/:routeParams?
```


### 完整路由地址

```fullroutes
/telegram/channel/zyfx_0315
```


### 相关文档

https://docs.rsshub.app/zh/routes/social-media#channel

### 预期是什么？

当设置了TELEGRAM_PROXY_HOST，TELEGRAM_PROXY_PORT，TELEGRAM_PROXY_SECRET后应该走MTProto协议正常获取内容
``

### 实际发生了什么？

TELEGRAM_PROXY_SECRET该参数只支持16 byte的密钥，而不支持更长的以ee开头的fake tls的密钥。

麻烦大佬处理一下，不然自建无法挂梯子

源码：`/lib/routes/telegram/tglib/client.ts` 21行
```
client = new TelegramClient(stringSession, apiId, apiHash, {
        connectionRetries: Infinity,
        autoReconnect: true,
        retryDelay: 3000,
        maxConcurrentDownloads: Number(config.telegram.maxConcurrentDownloads ?? 10),
        proxy:
            config.telegram.proxy?.host && config.telegram.proxy.port && config.telegram.proxy.secret
                ? {
                      ip: config.telegram.proxy.host,
                      port: Number(config.telegram.proxy.port),
                      MTProxy: true,
                      secret: config.telegram.proxy.secret,
                  }
                : undefined,
    });
```

### 部署

自建

### 部署相关信息

_No response_

### 额外信息

```shell
FetchError: FetchError: [GET] "https://t.me/s/zyfx_0315": <no response> fetch failed
at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
at async $fetchRaw2 (D:\my\dev\workspace_nodejs\RSSHub\node_modules\.pnpm\ofetch@1.4.1\node_modules\ofetch\dist\shared\ofetch.d8adb063.cjs:276:14)
at async $fetchRaw2 (D:\my\dev\workspace_nodejs\RSSHub\node_modules\.pnpm\ofetch@1.4.1\node_modules\ofetch\dist\shared\ofetch.d8adb063.cjs:276:14)
at async $fetch2 (D:\my\dev\workspace_nodejs\RSSHub\node_modules\.pnpm\ofetch@1.4.1\node_modules\ofetch\dist\shared\ofetch.d8adb063.cjs:322:15)
at async <anonymous> (D:\my\dev\workspace_nodejs\RSSHub\lib\routes\telegram\channel.ts:194:24)
at async Object.tryGet (D:\my\dev\workspace_nodejs\RSSHub\lib\utils\cache\index.ts:92:27)
at async Object.handler (D:\my\dev\workspace_nodejs\RSSHub\lib\routes\telegram\channel.ts:191:18)
at async wrappedHandler (D:\my\dev\workspace_nodejs\RSSHub\lib\registry.ts:96:33)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\cache.ts:57:9)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\parameter.ts:65:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\anti-hotlink.ts:101:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\header.ts:26:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\template.tsx:15:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\debug.ts:16:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\access-control.ts:21:9)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\sentry.ts:18:5)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\trace.ts:17:9)
at async dispatch (file:///D:/my/dev/workspace_nodejs/RSSHub/node_modules/.pnpm/hono@4.6.9/node_modules/hono/dist/compose.js:30:17)
at async middleware (D:\my\dev\workspace_nodejs\RSSHub\lib\middleware\logger.ts:36:5)
```


### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

请问telegram代理必须要搭这个MTProxy嘛。我用docker-compose部署在一台机器上，理论上不会有不同IP的情况。我尝试了全局透明代理，是可以获取到数据的，但是设置了PROXY_URI却报超时错误。

[2025-06-27T14:18:58.229] [ERROR] - [WebSocket connection failed attempt: 14]
Error: connect ETIMEDOUT 91.108.56.115:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1637:16) {
  errno: -110,
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '91.108.56.115',
  port: 80
}

--- a/package.json
+++ b/package.json
@@ -265,7 +265,8 @@
         },
         "patchedDependencies": {
             "@mswjs/interceptors@0.41.9": "patches/@mswjs__interceptors@0.41.9.patch",
-            "rss-parser@3.13.0": "patches/rss-parser@3.13.0.patch"
+            "rss-parser@3.13.0": "patches/rss-parser@3.13.0.patch",
+            "teleproto@1.228.5": "patches/teleproto@1.228.5.patch"
         }
     }
 }

--- a/patches/teleproto@1.228.5.patch
+++ b/patches/teleproto@1.228.5.patch
@@ -1,0 +1,84 @@
+diff --git a/network/connection/TCPMTProxy.js b/network/connection/TCPMTProxy.js
+index 80886d2db81dd8c8d4fbc5d57cc14b077f6fc754..13fdb8d73e737d40ad1c296568441158a0d787ea 100644
+--- a/network/connection/TCPMTProxy.js
++++ b/network/connection/TCPMTProxy.js
+@@ -3,7 +3,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
+ exports.ConnectionTCPMTProxyAbridged = exports.TCPMTProxy = void 0;
+ const node_crypto_1 = require("node:crypto");
+ const Connection_1 = require("./Connection");
+-const TCPAbridged_1 = require("./TCPAbridged");
++const errors_1 = require("../../errors");
+ const Helpers_1 = require("../../Helpers");
+ const CTR_1 = require("../../crypto/CTR");
+ const SECRET_LEN = 16;
+@@ -116,6 +116,7 @@ const TLS_RECORD_HANDSHAKE = 0x16;
+ const TLS_RECORD_CHANGE_CIPHER_SPEC = 0x14;
+ const TLS_RECORD_APPLICATION_DATA = 0x17;
+ const TLS_APP_DATA_PREFIX = Buffer.from([0x17, 0x03, 0x03]);
++const CLIENT_CHANGE_CIPHER_SPEC = Buffer.from([0x14, 0x03, 0x03, 0x00, 0x01, 0x01]);
+ const FAKE_TLS_HELLO_SIZE = 517;
+ const FAKE_TLS_HELLO_RANDOM_OFFSET = 11;
+ const FAKE_TLS_HELLO_RANDOM_LEN = 32;
+@@ -284,6 +285,7 @@ class FakeTlsSocket {
+         const hello = this.sendClientHello();
+         this.inner.write(hello);
+         await this.readAndVerifyServerHandshake();
++        this.inner.write(CLIENT_CHANGE_CIPHER_SPEC);
+     }
+     write(data) {
+         for (let off = 0; off < data.length; off += TLS_RECORD_MAX_PAYLOAD) {
+@@ -313,9 +315,9 @@ class FakeTlsSocket {
+         const hello = buildClientHello(this.domain, sessionId);
+         const stamp = (0, node_crypto_1.createHmac)("sha256", this.secret).update(hello).digest();
+         const now = Math.floor(Date.now() / 1000) >>> 0;
+-        const lastWord = (stamp.readUInt32BE(FAKE_TLS_RANDOM_TIMESTAMP_XOR_OFFSET) ^ now) >>>
++        const lastWord = (stamp.readUInt32LE(FAKE_TLS_RANDOM_TIMESTAMP_XOR_OFFSET) ^ now) >>>
+             0;
+-        stamp.writeUInt32BE(lastWord, FAKE_TLS_RANDOM_TIMESTAMP_XOR_OFFSET);
++        stamp.writeUInt32LE(lastWord, FAKE_TLS_RANDOM_TIMESTAMP_XOR_OFFSET);
+         stamp.copy(hello, FAKE_TLS_HELLO_RANDOM_OFFSET);
+         this.clientRandom = stamp;
+         return hello;
+@@ -398,14 +400,39 @@ class TCPMTProxy extends Connection_1.ObfuscatedConnection {
+     }
+ }
+ exports.TCPMTProxy = TCPMTProxy;
++class PaddedIntermediatePacketCodec extends Connection_1.PacketCodec {
++    constructor(props) {
++        super(props);
++        this.tag = PaddedIntermediatePacketCodec.tag;
++        this.obfuscateTag = PaddedIntermediatePacketCodec.obfuscateTag;
++    }
++    encodePacket(data) {
++        const padSize = (0, node_crypto_1.randomBytes)(1)[0] & 3;
++        const header = Buffer.alloc(4);
++        header.writeInt32LE(data.length + padSize, 0);
++        return Buffer.concat([header, data, (0, node_crypto_1.randomBytes)(padSize)]);
++    }
++    async readPacket(reader) {
++        const lengthBuffer = await reader.readExactly(4);
++        const length = lengthBuffer.readInt32LE(0);
++        const body = await reader.readExactly(length);
++        if (length === 4 && body.readInt32LE(0) < 0) {
++            throw new errors_1.InvalidBufferError(body);
++        }
++        const padSize = length % 4;
++        return padSize === 0 ? body : body.subarray(0, length - padSize);
++    }
++}
++PaddedIntermediatePacketCodec.tag = null;
++PaddedIntermediatePacketCodec.obfuscateTag = Buffer.from("dddddddd", "hex");
+ /**
+- * MTProxy connection using the Abridged packet codec — automatically selected
+- * when `proxy.MTProxy` is `true`.
++ * MTProxy connection using the padded intermediate packet codec — automatically
++ * selected when `proxy.MTProxy` is `true`.
+  */
+ class ConnectionTCPMTProxyAbridged extends TCPMTProxy {
+     constructor() {
+         super(...arguments);
+-        this.PacketCodecClass = TCPAbridged_1.AbridgedPacketCodec;
++        this.PacketCodecClass = PaddedIntermediatePacketCodec;
+     }
+ }
+ exports.ConnectionTCPMTProxyAbridged = ConnectionTCPMTProxyAbridged;

